rclcpp/src/rclcpp/qos.cpp

```diff
@@ -184,6 +185,35 @@ QoS::avoid_ros_namespace_conventions(bool avoid_ros_namespace_conventions)
   return *this;
 }
 
+namespace
+{
+/// Check if two rmw_time_t have the same values.
+bool operator==(const rmw_time_t & left, const rmw_time_t & right)
+{
+  return left.sec == right.sec && left.nsec == right.nsec;
+}
+}  // unnamed namespace
+
+bool operator==(const QoS & left, const QoS & right)
+{
+  const auto & pl = left.get_rmw_qos_profile();
+  const auto & pr = right.get_rmw_qos_profile();
+  return pl.history == pr.history &&
+         pl.depth == pr.depth &&
+         pl.reliability == pr.reliability &&
+         pl.durability == pr.durability &&
+         pl.deadline == pr.deadline &&
+         pl.lifespan == pr.lifespan &&
+         pl.liveliness == pr.liveliness &&
+         pl.liveliness_lease_duration == pr.liveliness_lease_duration &&
+         pl.avoid_ros_namespace_conventions == pr.avoid_ros_namespace_conventions;
```

Comment:
are the current tests covering all these individual conditions?

Reply:
no, but it's covering every type of comparison (enum, rmw_time_t, int). I could go in and add a check for each member

Reply:
the tests now cover every condition